app/gui/suggestion-database/src/documentation_ir.rs

```diff
@@ -75,51 +105,155 @@ impl EntryDocumentation {
         Ok(result)
     }
 
-    /// Qualified name of the function-like entry. See [`Documentation::function_name`].
-    pub fn function_name(&self) -> Option<&QualifiedName> {
+    /// Create documentation for a hard-coded builtin entry.
+    pub fn builtin(sections: impl IntoIterator<Item = &DocSection>) -> Self {
+        let sections = BuiltinDocumentation::from_doc_sections(sections.into_iter());
+        Self::Docs(Documentation::Builtin(sections))
+    }
+
+    /// The list of links displayed on the documentation page.
+    pub fn linked_doc_pages(&self) -> Vec<LinkedDocPage> {
         match self {
-            EntryDocumentation::Docs(docs) => docs.function_name(),
-            _ => None,
+            EntryDocumentation::Docs(docs) => match docs {
+                // Module documentation contains links to all methods and types defined in this
+                // module.
+                Documentation::Module(docs) => {
+                    let methods = docs.methods.iter().map(|method| LinkedDocPage {
+                        name: method.name.clone_ref(),
+                        page: Documentation::ModuleMethod {
+                            docs:        method.clone_ref(),
+                            module_docs: docs.clone_ref(),
```

Comment:
We could make it a lot shorter in many places by doing `Documentation::ModuleMethod(&method, &module_docs)` and define its parameters in the `new` fn as `something: Into<...>`, This way it will work both for references and passing owned values, making the usage way nicer.